packages/analytics-browser/src/browser-client.ts

```diff
@@ -43,17 +47,31 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
     }
     this.initializing = true;
 
-    // Step 1: Read cookies stored by old SDK
-    const oldCookies = await parseOldCookies(options.apiKey, options);
+    // Step 1: Read cookies stored by SDK
+    options.domain = options.disableCookies ? '' : options.domain ?? (await getTopLevelDomain());
+    const legacyCookies = await parseLegacyCookies(options.apiKey, options);
+    const cookieStorage = await createCookieStorage<UserSession>(options);
+    const previousCookies = await cookieStorage.get(getCookieName(options.apiKey));
+    const queryParams = getQueryParams();
 
     // Step 2: Create browser config
+    const deviceId = options.deviceId ?? queryParams.deviceId ?? previousCookies?.deviceId ?? legacyCookies.deviceId;
+    const sessionId = options.sessionId ?? previousCookies?.sessionId ?? legacyCookies.sessionId;
+    const optOut = options.optOut ?? previousCookies?.optOut ?? legacyCookies.optOut;
+    const lastEventTime = previousCookies?.lastEventTime ?? legacyCookies.lastEventTime;
+    const userId = options.userId ?? previousCookies?.userId ?? legacyCookies.userId;
+
+    this.previousSessionDeviceId = previousCookies?.deviceId ?? legacyCookies.deviceId;
+    this.previousSessionUserId = previousCookies?.userId ?? legacyCookies.userId;
```

Comment:
Pull out reading of cookies to client level to store `previousSessionDeviceId` and `previousSessionUserId`. This is used for sending the first session end with those IDs